src/Custom/Microsoft.Extensions.AI.OpenAIAdapter.cs

```diff
@@ -0,0 +1,1820 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Runtime.CompilerServices;
+using System.Text;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using System.Threading;
+using System.Threading.Tasks;
+using OpenAI;
+using OpenAI.Audio;
+using OpenAI.Chat;
+using OpenAI.Embeddings;
+using OpenAI.Responses;
+
+#nullable enable
+
+namespace Microsoft.Extensions.AI;
+
+/// <summary>Represents an <see cref="IChatClient"/> for an OpenAI <see cref="OpenAIClient"/> or <see cref="ChatClient"/>.</summary>
+internal sealed partial class OpenAIChatClient : IChatClient
+{
+    /// <summary>Gets the default OpenAI endpoint.</summary>
+    private static Uri DefaultOpenAIEndpoint { get; } = new("https://api.openai.com/v1");
+
+    /// <summary>Metadata about the client.</summary>
+    private readonly ChatClientMetadata _metadata;
+
+    /// <summary>The underlying <see cref="ChatClient" />.</summary>
+    private readonly ChatClient _chatClient;
+
+    /// <summary>Initializes a new instance of the <see cref="OpenAIChatClient"/> class for the specified <see cref="ChatClient"/>.</summary>
+    /// <param name="chatClient">The underlying client.</param>
+    /// <exception cref="ArgumentNullException"><paramref name="chatClient"/> is <see langword="null"/>.</exception>
+    public OpenAIChatClient(ChatClient chatClient)
+    {
+        _ = Throw.IfNull(chatClient);
+
+        _chatClient = chatClient;
+
+        // https://github.com/openai/openai-dotnet/issues/215
+        // The endpoint and model aren't currently exposed, so use reflection to get at them, temporarily. Once packages
+        // implement the abstractions directly rather than providing adapters on top of the public APIs,
+        // the package can provide such implementations separate from what's exposed in the public API.
+        Uri providerUrl = typeof(ChatClient).GetField("_endpoint", BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Instance)
```

Comment:
Now that it's in the same assembly, it should not be using reflection. (That goes for here and anywhere else reflection is being used.)